src/schema/posts.ts

```diff
@@ -1101,6 +1219,44 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
   BaseContext
 >({
   Query: {
+    squadPostModerationsBySourceId: async (
+      _,
+      { sourceId }: { sourceId: string },
+      ctx: Context,
+      info,
+    ): Promise<GQLSquadPostModeration[]> => {
+      const isModerator = await isPrivilegedMember(ctx, sourceId);
```

Comment:
you can use `ensureSourcePermission` here by adding a new permission to `moderatorPermissions` and `adminPermissions`. That way you don't need to create new `isPrivilegedMember` function

Reply:
I'll look into this

Reply:
@capJavert I don't think using ensureSourcePermission is the right call here, mainly because it throws error if they don't have permission, which would make the moderation required step a bit awkward. I think this little isPrivilegedMember function works well.

However, I will redo my change in boot, and remove the role property that I added, and instead add a permission "Moderate", to check what mutation to call from the frontend!

Reply:
Ok, we should still check however that user can't create a post directly if its not allowed without moderation on the squad eg. someone just takes the API calls and does them manually through postman.

Reply:
See this is done in `canPostToSquad` 🟢